File: sources/CSharp/0.6.0-feature-ers-925-6279-3364fd8f/src/Ers/System/PathAnimationSystem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ers.Engine;

namespace Ers
{
    public static class PathAnimationSystem
    {
        public static void Animate(Entity toAnimate, SimulationTime duration, float fromValue, float toValue, Entity entityContainingPath, int pathIndex)
        {
            SimulationTime currentTime = SubModel.GetSubModel().GetSimulator().GetCurrentTime();
            ErsEngine.ERS_PathAnimationSystem_Animate(toAnimate, currentTime, currentTime+duration, fromValue, toValue, entityContainingPath, pathIndex);
        }

        public static void Update(SimulationTime currentTime)
        {
            ErsEngine.ERS_PathAnimationSystem_Update(currentTime);
        }
    }
}

File: sources/CPP/include/Ers/Systems/PathAnimationSystem.cpp
#include "PathAnimationSystem.h"

#include "Ers/Api.h"
#include "Ers/Model/Simulator/Simulator.h"
#include "Ers/SubModel/SubModel.h"

namespace Ers
{
    void PathAnimationSystem::Animate(
        EntityID entity, SimulationTime duration, float fromValue, float toValue, Entity entityContainingPath, int pathIndex)
    {
        SimulationTime currentTime = GetSubModel().GetSimulator().CurrentTime();
        ersAPIFunctionPointers.ERS_PathAnimationSystem_Animate(
            entity, currentTime, duration, fromValue, toValue, entityContainingPath, pathIndex);
    }

    void PathAnimationSystem::Update(SimulationTime currentTime)
    {
        ersAPIFunctionPointers.ERS_PathAnimationSystem_Update(currentTime);
    }
} // namespace Ers
